### PR TITLE
Fixed #30157 -- Dropped support for Oracle 12.1.

### PR DESCRIPTION
**Pull request by felixxm**
Ticket [30157](https://code.djangoproject.com/ticket/30157).

### CODE REVIEW

**Comment by felixxm** [comment]
buildbot, test on oracle.

File: django/db/backends/oracle/features.py [hunk 1]
```diff
@@ -1,10 +1,10 @@
 from django.db.backends.base.features import BaseDatabaseFeatures
 from django.db.utils import InterfaceError
-from django.utils.functional import cached_property
 
 
 class DatabaseFeatures(BaseDatabaseFeatures):
     interprets_empty_strings_as_nulls = True
+    allow_sliced_subqueries_with_in = True
```
Comment (timgraham) [inline django/db/backends/oracle/features.py]: Unneeded since it's the default.

File: docs/releases/3.0.txt [hunk 2]
```diff
@@ -235,6 +235,12 @@ Dropped support for PostgreSQL 9.4
 Upstream support for PostgreSQL 9.4 ends in December 2019. Django 3.0 supports
 PostgreSQL 9.5 and higher.
 
+Dropped support for Oracle 12.1
+-------------------------------
+
+Upstream support for Oracle 12.1 ends in July 2021. Django 2.2 will be
+supported until April 2022. Django 3.0 officially supports Oracle 18c.
```
Comment (timgraham) [inline docs/releases/3.0.txt]: As someone unfamiliar with Oracle version number, it's unclear to me what the status of Oracle 12.2 is?

**Comment by felixxm** [comment]
@timgraham Thanks for the review :+1:. I'm going to send separate PR with changing links to the Oracle 18c doc.

**Pull request merged**